---
title: "<center> HNM RNAseq analysis <center>"
author: "<center> Felix Westcott <center><br>"
date: "<center> _`r Sys.Date()`_ <center>"
---

```{r setup, include=FALSE}

library(tidyverse)
library(edgeR)
library(limma)
library(FactoMineR)
library(factoextra)
library(cowplot)
library(gt)
library(DT)
library(ggplot2)
library(EnhancedVolcano)
library(ggpubr)
library(gplots)
library(venn)
library(patchwork)
library(GenomeInfoDbData)
library(org.Hs.eg.db)
library(clusterProfiler)
library(pathview)
library(ggsci)
library(ggprism)
library(Glimma)
library(fgsea)
library(readxl)

```

``` {R Loading data}
# Load data
Raw_counts <- read_excel("gene_count.xlsx", sheet = 1)
samplemetadata <- read_excel('Sample_info.xlsx', sheet = 1)

group <- factor(samplemetadata$group, levels = c("LC", "LO", "LP", "HC", "HO", "HP"))
exp <- factor(samplemetadata$exp)
insulin <- factor(samplemetadata$insulin)
fat <- factor(samplemetadata$fat)

# Turn raw count data into matrix with only relevant parts
geneID <- Raw_counts$gene_name
Raw_counts <- as.matrix(Raw_counts[,2:37])
rownames(Raw_counts) <- geneID

# Create a DGEList from raw count matrix
myDGEList <- DGEList(Raw_counts, group = group)

cpm <- cpm(myDGEList, log=FALSE)
log2.cpm <- cpm(myDGEList, log=TRUE)

log2.cpm <- as_tibble(log2.cpm, rownames = "geneID")
log2.cpm.pivot <- pivot_longer(log2.cpm,
                                  cols = LC1:HP6, 
                                  names_to = "samples",
                                  values_to = "expression")

ggplot(log2.cpm.pivot) +
  aes(x=samples, y=expression, fill=samples) +
  geom_violin(trim = FALSE, show.legend = FALSE) +
  stat_summary(fun = "median", 
               geom = "point", 
               shape = 95, 
               size = 10, 
               color = "black", 
               show.legend = FALSE) +
  labs(y="log2 expression", x = "sample",
       title="Log2 Counts per Million (CPM)",
       subtitle="unfiltered, non-normalized") +
  theme_bw() +
   theme(axis.text.x = element_text(size = 7))
```

``` {R Filter and normalise data}

# Filter data. NB this uses cpm but this will get calculated properly again later

keep <- filterByExpr(cpm, group = group)
myDGEList.filtered <- myDGEList[keep,,keep.lib.sizes=FALSE]

# Normalize filtered data using TMM method
myDGEList.filtered.norm <- calcNormFactors(myDGEList.filtered, method = "TMM")

# Express filtered and normalised data in log2(cpm)
log2.cpm.filtered.norm <- cpm(myDGEList.filtered.norm, log=TRUE)

# Also have it as a dataframe in case it comes in handy
log2.cpm.filtered.norm.df <- as_tibble(log2.cpm.filtered.norm, rownames = "geneID")

log2.cpm.filtered.norm.df.pivot <- pivot_longer(log2.cpm.filtered.norm.df, 
                                                cols = LC1:HP6, 
                                                names_to = "samples", 
                                                values_to = "expression")

ggplot(log2.cpm.filtered.norm.df.pivot) +
  aes(x=samples, y=expression, fill=samples) +
  geom_violin(trim = FALSE, show.legend = FALSE) +
  stat_summary(fun = "median", 
               geom = "point", 
               shape = 95, 
               size = 10, 
               color = "black", 
               show.legend = FALSE) +
  labs(y="log2 expression", x = "sample",
       title="Log2 Counts per Million (CPM)",
       subtitle="filtered, TMM normalized") +
  theme_bw() +
   theme(axis.text.x = element_text(size = 7))

```

``` {R Gene Expression Data}

# Creating a log2cpm gene expression df with averages for each condition
Avg.gene_expression <- log2.cpm.filtered.norm.df %>% 
  mutate(LC = (LC1 + LC2 + LC3 + LC4 + LC5 + LC6)/6,
         LO = (LO1 + LO2 + LO3 + LO4 + LO5 + LO6)/6,
         LP = (LP1 + LP2 + LP3 + LP4 + LP5 + LP6)/6,
         HC = (HC1 + HC2 + HC3 + HC4 + HC5 + HC6)/6,
         HO = (HO1 + HO2 + HO3 + HO4 + HO5 + HO6)/6,
         HP = (HP1 + HP2 + HP3 + HP4 + HP5 + HP6)/6) %>% 
    mutate_if(is.numeric, round, 2) %>%
  dplyr::select(geneID, LC, LO, LP, HC, HO, HP)

datatable(Avg.gene_expression, 
          extensions = c('KeyTable', "FixedHeader"), 
          filter = 'top',
          caption = 'Table 1: log2 CPM of filtered and normalised genes',
          options = list(keys = TRUE, 
                         searchHighlight = TRUE, 
                         pageLength = 10, 
                         lengthMenu = c("10", "25", "50", "100")))

```

```{R, Really cool PCA browser pop out}

myDGEList.filtered.norm$samples$exp <- exp
myDGEList.filtered.norm$samples$insulin <- insulin
myDGEList.filtered.norm$samples$fat <- fat

glMDSPlot(log2.cpm.filtered.norm, labels=paste(exp, insulin, fat, sep="_"), groups=myDGEList.filtered.norm$samples[,c(1,4,5,6)], launch=TRUE)

```


``` {R PCA analysis and scree plot}
#This is exactly the same as before, I just added the summary print out

pca.res <- prcomp(t(log2.cpm.filtered.norm), scale.=F, retx=T)

summary(pca.res)

fviz_eig(pca.res, addlabels = TRUE, ylim = c(0,100))

```

``` {R individual PCA plot}
#This is exactly the same as before I think

fviz_pca_ind(pca.res, geom.ind = "point", pointshape = 21, 
             pointsize = 2, 
             fill.ind = exp,
             palette = "npg",
             col.ind = "black", 
             addEllipses = TRUE,
             label = "var",
             col.var = "black",
             repel = TRUE,
             legend.title = "Group", title = "Principle Components Analysis") +
  theme_prism(base_size = 14)

#ggsave("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Outputs/Transfer-report/Figures/PCA.png")

```


``` {R PCA 'small mulitples' plots}

# This is all new so I'll explain what I'm doing.

# Take results of first four PCAs and turn into tibble with sample names and fat types as added rows

pca.res.df <- pca.res$x[,1:4] %>%
  as_tibble() %>%
  add_column(sample = samplemetadata$sample_ID,
             group = group)
  
#Pivot the table to make it tidy and easier to facet graph

pca.pivot <- pivot_longer(pca.res.df, 
                          cols = PC1:PC4, 
                          names_to = "PC", 
                          values_to = "loadings")

#Graph - this graph illustrates if there are any obvious confounding variables among your samples (ie batch effects etc.)

ggplot(pca.pivot) +
  aes(x=sample, y=loadings, fill=group) + 
  geom_bar(stat="identity") +
  facet_wrap(~PC) +
  labs(title="PCA 'small multiples' plot") +
  theme_bw() +
  coord_flip() +
theme(axis.text.x = element_text(color = "black", size = 8, angle = 0, hjust = .5, vjust = .5, face = "plain"),
      axis.text.y = element_text(color = "black", size = 8, angle = 0, hjust = .5, vjust = .5, face = "plain"),
      axis.title.x = element_text(color = "black", size = 12, angle = 0, hjust = .5, vjust = .5, face = "plain"),
      axis.title.y = element_text(color = "black", size = 12, angle = 90, hjust = .5, vjust = .5, face = "plain"))

```

```  {R, Mixed Model}

#OK there's a lot here to explain so listen up

#First of all we make a design matrix for the purposes of this we only care about the fixed effect of "group" variable which codes both fat conc and composition combined into one variable. The 0 at the front of the matrix model formula shows that we do not want an intercept because we want to compare all factors in group to each other and not just to control and it's a lot easier to code for this when there's no intercept (but otherwise makes no difference to the actual model as group is a factor). 

design <- model.matrix(~0+group)
colnames(design) <- gsub("group", "", colnames(design))
design

#We then make our contrast matrix which dictates all the contrasts we will be interested in, simple as that.
contrasts <- makeContrasts(
   OvC = (LO + HO)/2 - (LC + HC)/2,
   PvC = (LP + HP)/2 - (LC + HC)/2,
   PvO = (LP + HP)/2 - (LO + HO)/2,
   HvL = (HC + HO + HP)/3 - (LC + LO + LP)/3,
   HCvLC = HC - LC,
   HOvLO = HO - LO,
   HPvLP = HP - LP,
   LOvLC = LO - LC,
   LPvLC = LP - LC,
   LPvLO = LP - LO,
   HOvHC = HO - HC,
   HPvHC = HP - HC,
   HPvHO = HP - HO,
   HPvLC = HP - LC,
   levels = colnames(design))
rownames(contrasts) <- gsub("group", "", rownames(contrasts))
contrasts

# The modelling that Limma uses assumes the counts are normally distributed which they usually are not so this is us removing heteroscedascity from count data. voom function is usually used but here we use voomWithualityWeights because we know there's a huge amount of variability in our samples. It produces a "before" graphs to show just how poorly distributed they are.

v <- voomWithQualityWeights(myDGEList.filtered.norm, design, plot=TRUE)

#We're going to start creating a mixed effect model whereby "exp" is a random effects factor (ie a factor we don't care about but has an impact on the data) and "group" is the fixed effect factor. We begin by generating a correlation coefficient for duplicate measurements made on the same exp.

cor <- duplicateCorrelation(v, design, block=exp)
cor$consensus.correlation

#We run the voom again as the correlation will change the weights slightly

v <- voomWithQualityWeights(myDGEList.filtered.norm, design, plot = TRUE, block = exp, correlation = cor$consensus)

#And hence we have to run the correlation again now the voom has changed (but it doesn't make much difference)

cor <- duplicateCorrelation(v, design, block=exp)
cor$consensus.correlation

#Now we can actually make the model, and carry out the contrasts analysis as all coded. NB - we use a mixed model (exp = random effect factor) rather than linear model (exp = fixed effect factor) to save on degrees of freedom and give greater statistical power.

vfit <- lmFit(object=v, design=design, 
  block=exp, correlation=cor$consensus.correlation)
vfit <- contrasts.fit(vfit, contrasts=contrasts)
efit1 <- eBayes(vfit)
tfit1 <- treat(vfit, lfc=0)
plotSA(tfit1)

# references: 
# https://f1000research.com/articles/5-1408/v3
# https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html section 7.6
# https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf section 18.1.9
```

```{R, Model comparison}
#Mixed model
dt1 <- decideTests(tfit1)

summary(dt1)

```

```{R, Venns}

vennDiagram(dt1[,1:2], circle.col=c("turquoise", "salmon"))

vennDiagram(dt1[,5:7], circle.col=c("green", "red", "blue"))

vennDiagram(dt1[,c(9,5,8)], circle.col=c("green", "red", "blue"))

vennDiagram(dt1[,c(14,2,4)], circle.col=c("green", "red", "blue"))


```

```{R, Volcano plots!}

OvC <- topTreat(tfit1, coef=1, n=Inf)
PvC <- topTreat(tfit1, coef=2, n=Inf)
PvO <- topTreat(tfit1, coef=3, n=Inf)
HvL <- topTreat(tfit1, coef=4, n=Inf)
HPvLC <- topTreat(tfit1, coef=14, n =Inf)

# Almost exactly the same just all in one place and made to look better 

EnhancedVolcano(OvC,
    lab = OvC$ID,
    x = 'logFC',
    y = 'adj.P.Val',
    legendPosition = "right",
    legendLabSize = 12,
    legendIconSize = 3,
    labSize = 3,
    subtitle = "OPLA vs Control",
    pCutoff = 0.05, 
    FCcutoff = 0.2)


EnhancedVolcano(PvC,
    lab = PvC$ID,
    x = 'logFC',
    y = 'adj.P.Val',
    legendPosition = "right",
    legendLabSize = 12,
    legendIconSize = 3,
    labSize = 3,
    subtitle = "POLA vs Control",
    pCutoff = 0.05, 
    FCcutoff = 0.2)


EnhancedVolcano(PvO,
    lab = PvO$ID,
    x = 'logFC',
    y = 'adj.P.Val',
    legendPosition = "right",
    legendLabSize = 12,
    legendIconSize = 3,
    labSize = 3,
    subtitle = "POLA vs OPLA",
    pCutoff = 0.05, 
    FCcutoff = 0.2)

EnhancedVolcano(HvL,
    lab = HvL$ID,
    x = 'logFC',
    y = 'adj.P.Val',
    legendPosition = "right",
    legendLabSize = 12,
    legendIconSize = 3,
    labSize = 3,
    subtitle = "High Insulin vs Low Insulin",
    pCutoff = 0.05, 
    FCcutoff = 0.2)

#Interactive volcano plot for HF v LF (change the columns of the data frames to make it for other comparisons)
glXYPlot(x=efit1$coef[,3], y=efit1$lods[,3], xlab="logFC", ylab="logP", status=dt1[,3], counts=log2.cpm.filtered.norm, groups=group, launch = TRUE, sample.cols=group)

#glimmaVolcano(tfit1, coef = 1, status = dt1, main=colnames(tfit1)[1], side.main="ID", counts=log2.cpm.filtered.norm, transform = "none", groups=group, launch=TRUE)

#glMDPlot(tfit1, coef=4, status=dt1, main=colnames(tfit1)[4], side.main="ID", counts=log2.cpm.filtered.norm, groups=group, launch=TRUE)

```

``` {R, Table of differential gene expression}

dt1 <- as.data.frame(dt1)

datatable(dt1, 
          extensions = c('KeyTable', "FixedHeader"), 
          options = list(keys = TRUE, searchHighlight = TRUE, pageLength = 10, lengthMenu = c("10", "25", "50", "100")))

```

``` {R Graph Individual Genes}

# Pivot to tidy up data
Avg.gene_expression.pivot <- pivot_longer(Avg.gene_expression, 
                          cols = LC : HP, 
                          names_to = "group", 
                          values_to = "lcpm")

#  Filter for genes you want. grepl will pick out any with that phrase in it. Use " x|y|z|etc." notation to search for multiple genes. 
graph <- Avg.gene_expression.pivot %>%
  dplyr::filter(grepl('MAPK8|MAPK9', geneID))

# Graph error bars were a step too far for me as they got messed up with the facet wrap
ggbarplot(graph, "group", "lcpm", add = "mean_se", palette = "npg", legend = "none", fill = "group", xlab = "Media", ylab = "counts per million (log2)") +
  facet_wrap(~geneID) + theme_prism(base_size = 14) +
  theme(legend.position = "none", legend.title = element_text()) +
  scale_x_discrete(labels = function(x) str_wrap(x, width = 1)) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.22)))
#ggsave('C:/Users/fwestcott/OneDrive - Nexus365/Desktop/RNAseq Graph2.png')

```

``` {R Heatmap of selected genes}

# Change whatever is in the filter function to change which genes are plotted. use the form 'gene1|gene2' etc.
selected_genes <- Avg.gene_expression %>%
dplyr::filter(grepl('ATG10|ATG14|ATG16L1|ATG3|ATG4|ATG7|ATG9A|BCL2|GABARAP|GABARAPL1|MAP1LC3B|NBR1|OPTN|PINK1|SQSTM1|ULK1|ULK2|UVRAG|WDR45|WIPI1',geneID))%>%
 column_to_rownames(var="geneID") %>%
  as.matrix

#png(filename = "C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/OPLA-POLA-Primary-RNAseq/ATG heatmap.png", width = 800, height = 800, units = "px", pointsize = 17, bg = "white")

heatmap.2(selected_genes, scale="row",
  labRow=rownames(selected_genes), labCol=colnames(selected_genes),
   trace="none", density.info="none", 
    margin=c(10,20), lhei=c(2,10), dendrogram="column")

#dev.off()

```


```{R, OvC Functional enrichment analysis}

diffGenes.OvC <- rownames(filter(dt1, dt1$OvC != 0))

diffGenes.OvC <- OvC[OvC$ID %in% diffGenes.OvC, ] %>%
rename("SYMBOL" = ID)

diffGenes.OvC <- diffGenes.OvC %>% 
    dplyr::select(SYMBOL, t) %>% 
    na.omit() %>% 
    distinct() %>% 
    group_by(SYMBOL) %>% 
    summarize(stat=mean(t))

ranks <- deframe(diffGenes.OvC)

# Load the pathways into a named list
pathways.hallmark <- gmtPathways("h.all.v2023.1.Hs.symbols.gmt")

fgseaRes <- fgsea(pathways=pathways.hallmark, stats=ranks, nperm=1000)

fgseaResTidy <- fgseaRes %>%
  as_tibble() %>%
  arrange(desc(NES))

# Show in a nice table:
fgseaResTidy %>% 
  dplyr::select(-leadingEdge, -ES, -nMoreExtreme) %>% 
  arrange(padj) %>% 
  DT::datatable()

ggplot(fgseaResTidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

#KEGG
diffGenes.OvC <- rownames(filter(dt1, dt1$OvC != 0))

diffGenes.OvC <- OvC[OvC$ID %in% diffGenes.OvC, ] %>%
rename("SYMBOL" = ID)

entrez <- bitr(diffGenes.OvC$SYMBOL, fromType = "SYMBOL",
                 toType = "ENTREZID",
                 OrgDb="org.Hs.eg.db")

deg_OvC_enrich <- diffGenes.OvC %>% left_join(entrez,.,by="SYMBOL")

OvC_kegg_gene_list <- deg_OvC_enrich$logFC

names(OvC_kegg_gene_list) <- deg_OvC_enrich$ENTREZID

OvC_kegg_gene_list<-na.omit(OvC_kegg_gene_list)

OvC_kegg_gene_list = sort(OvC_kegg_gene_list, decreasing = TRUE)

OvC_kk <- gseKEGG(geneList     = OvC_kegg_gene_list,
               organism     = "hsa")

dotplot(OvC_kk, showCategory = 5, title = "OvC Enriched Pathways vs Control" , split=".sign") + facet_grid(.~.sign)
 # ggsave("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/OPLA-POLA-Primary-RNAseq/OvC GSEA.png")

```


``` {R OvC pathway enrichment, warning=FALSE}
setwd("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/HNM-huh7-RNAseq/Pathview")

#This is the same as before
fc.vector <- as.numeric(deg_OvC_enrich[, 3])
names(fc.vector) <- (deg_OvC_enrich[, 2])
de.names <- names(fc.vector) 

pathview(gene.data  = fc.vector, pathway.id = "hsa04932",species  = "hsa")

```

``` {R PvC function enrichment analysis}
#MSigDb Hallmarks
diffGenes.PvC <- rownames(filter(dt1, dt1$PvC != 0))

diffGenes.PvC <- PvC[PvC$ID %in% diffGenes.PvC, ] %>%
rename("SYMBOL" = ID)

diffGenes.PvC <- diffGenes.PvC %>% 
    dplyr::select(SYMBOL, t) %>% 
    na.omit() %>% 
    distinct() %>% 
    group_by(SYMBOL) %>% 
    summarize(stat=mean(t))

ranks <- deframe(diffGenes.PvC)

pathways.hallmark <- gmtPathways("h.all.v2023.1.Hs.symbols.gmt")

fgseaRes <- fgsea(pathways=pathways.hallmark, stats=ranks, nperm=1000)

fgseaResTidy <- fgseaRes %>%
  as_tibble() %>%
  arrange(desc(NES))

fgseaResTidy %>% 
  dplyr::select(-leadingEdge, -ES, -nMoreExtreme) %>% 
  arrange(padj) %>% 
  DT::datatable()

ggplot(fgseaResTidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

#Kegg
diffGenes.PvC <- rownames(filter(dt1, dt1$PvC != 0))

diffGenes.PvC <- PvC[PvC$ID %in% diffGenes.PvC, ] %>%
rename("SYMBOL" = ID)

entrez <- bitr(diffGenes.PvC$SYMBOL, fromType = "SYMBOL",
                 toType = "ENTREZID",
                 OrgDb="org.Hs.eg.db")

deg_PvC_enrich <- diffGenes.PvC %>% left_join(entrez,.,by="SYMBOL")

PvC_kegg_gene_list <- deg_PvC_enrich$logFC

names(PvC_kegg_gene_list) <- deg_PvC_enrich$ENTREZID

PvC_kegg_gene_list<-na.omit(PvC_kegg_gene_list)

PvC_kegg_gene_list = sort(PvC_kegg_gene_list, decreasing = TRUE)

PvC_kk <- gseKEGG(geneList     = PvC_kegg_gene_list,
               organism     = "hsa")

dotplot(PvC_kk, showCategory = 5, title = "PvC Enriched Pathways vs Control" , split=".sign") + facet_grid(.~.sign)
 # ggsave("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/OPLA-POLA-Primary-RNAseq/PvC GSEA.png")
```


``` {R PvC pathway enrichment, warning=FALSE}
setwd("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/HNM-huh7-RNAseq/Pathview")

#This is the same as before
fc.vector <- as.numeric(deg_PvC_enrich[, 3])
names(fc.vector) <- (deg_PvC_enrich[, 2])
de.names <- names(fc.vector)

pathview(gene.data  = fc.vector, pathway.id = "hsa04932",species  = "hsa")

```


``` {R HPvLC function enrichment analysis}
#MSigDb Hallmarks
diffGenes.HPvLC <- rownames(filter(dt1, dt1$HPvLC != 0))

diffGenes.HPvLC <- HPvLC[HPvLC$ID %in% diffGenes.HPvLC, ] %>%
rename("SYMBOL" = ID)

diffGenes.HPvLC <- diffGenes.HPvLC %>% 
    dplyr::select(SYMBOL, t) %>% 
    na.omit() %>% 
    distinct() %>% 
    group_by(SYMBOL) %>% 
    summarize(stat=mean(t))

ranks <- deframe(diffGenes.HPvLC)

# Load the pathways into a named list
pathways.hallmark <- gmtPathways("h.all.v2023.1.Hs.symbols.gmt")

fgseaRes <- fgsea(pathways=pathways.hallmark, stats=ranks, nperm=1000)

fgseaResTidy <- fgseaRes %>%
  as_tibble() %>%
  arrange(desc(NES))

# Show in a nice table:
fgseaResTidy %>% 
  dplyr::select(-leadingEdge, -ES, -nMoreExtreme) %>% 
  arrange(padj) %>% 
  DT::datatable()

ggplot(fgseaResTidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

#Kegg
diffGenes.HPvLC <- rownames(filter(dt1, dt1$HPvLC != 0))

diffGenes.HPvLC <- HPvLC[HPvLC$ID %in% diffGenes.HPvLC, ] %>%
rename("SYMBOL" = ID)

entrez <- bitr(diffGenes.HPvLC$SYMBOL, fromType = "SYMBOL",
                 toType = "ENTREZID",
                 OrgDb="org.Hs.eg.db")

deg_HPvLC_enrich <- diffGenes.HPvLC %>% left_join(entrez,.,by="SYMBOL")

HPvLC_kegg_gene_list <- deg_HPvLC_enrich$logFC

names(HPvLC_kegg_gene_list) <- deg_HPvLC_enrich$ENTREZID

HPvLC_kegg_gene_list<-na.omit(HPvLC_kegg_gene_list)

HPvLC_kegg_gene_list = sort(HPvLC_kegg_gene_list, decreasing = TRUE)

HPvLC_kk <- gseKEGG(geneList     = HPvLC_kegg_gene_list,
               organism     = "hsa")

dotplot(HPvLC_kk, showCategory = 5, title = "HPvLC Enriched Pathways vs Control" , split=".sign") + facet_grid(.~.sign)
 # ggsave("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/OPLA-POLA-Primary-RNAseq/HPvLC GSEA.png")
```


``` {R HPvLC pathway enrichment, warning=FALSE}
setwd("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/HNM-huh7-RNAseq/Pathview")

#This is the same as before
fc.vector <- as.numeric(deg_HPvLC_enrich[, 3])
names(fc.vector) <- (deg_HPvLC_enrich[, 2])
de.names <- names(fc.vector)

pathview(gene.data  = fc.vector, pathway.id = "hsa04932",species  = "hsa")

```

``` {R PvO function enrichment analysis}
#MSigDb
diffGenes.PvO <- rownames(filter(dt1, dt1$PvO != 0))

diffGenes.PvO <- PvO[PvO$ID %in% diffGenes.PvO, ] %>%
rename("SYMBOL" = ID)

diffGenes.PvO <- diffGenes.PvO %>% 
    dplyr::select(SYMBOL, t) %>% 
    na.omit() %>% 
    distinct() %>% 
    group_by(SYMBOL) %>% 
    summarize(stat=mean(t))

ranks <- deframe(diffGenes.PvO)

# Load the pathways into a named list
pathways.hallmark <- gmtPathways("h.all.v2023.1.Hs.symbols.gmt")

fgseaRes <- fgsea(pathways=pathways.hallmark, stats=ranks, nperm=1000)

fgseaResTidy <- fgseaRes %>%
  as_tibble() %>%
  arrange(desc(NES))

# Show in a nice table:
fgseaResTidy %>% 
  dplyr::select(-leadingEdge, -ES, -nMoreExtreme) %>% 
  arrange(padj) %>% 
  DT::datatable()

ggplot(fgseaResTidy, aes(reorder(pathway, NES), NES)) +
  geom_col(aes(fill=padj<0.05)) +
  coord_flip() +
  labs(x="Pathway", y="Normalized Enrichment Score",
       title="Hallmark pathways NES from GSEA") + 
  theme_minimal()

#Kegg 
diffGenes.PvO <- rownames(filter(dt1, dt1$PvO != 0))

diffGenes.PvO <- PvO[PvO$ID %in% diffGenes.PvO, ] %>%
rename("SYMBOL" = ID)

entrez <- bitr(diffGenes.PvO$SYMBOL, fromType = "SYMBOL",
                 toType = "ENTREZID",
                 OrgDb="org.Hs.eg.db")

deg_PvO_enrich <- diffGenes.PvO %>% left_join(entrez,.,by="SYMBOL")

PvO_kegg_gene_list <- deg_PvO_enrich$logFC

names(PvO_kegg_gene_list) <- deg_PvO_enrich$ENTREZID

PvO_kegg_gene_list<-na.omit(PvO_kegg_gene_list)

PvO_kegg_gene_list = sort(PvO_kegg_gene_list, decreasing = TRUE)

PvO_kk <- gseKEGG(geneList     = PvO_kegg_gene_list,
               organism     = "hsa")

dotplot(PvO_kk, showCategory = 10, title = "PvO Enriched Pathways vs Control" , split=".sign") + facet_grid(.~.sign)
 # ggsave("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/OPLA-POLA-Primary-RNAseq/PvO GSEA.png")
```


``` {R PvO pathway enrichment, warning=FALSE}
setwd("C:/Users/fwestcott/OneDrive - Nexus365/Desktop/Results/HNM-huh7-RNAseq/Pathview")

#This is the same as before
fc.vector <- as.numeric(deg_PvO_enrich[, 3])
names(fc.vector) <- (deg_PvO_enrich[, 2])
de.names <- names(fc.vector)

pathview(gene.data  = fc.vector, pathway.id = "hsa04932",species  = "hsa")

```